tools/compile-compose.c

```diff
@@ -69,10 +147,17 @@ print_compose_table_entry(struct xkb_compose_table_entry *entry)
             printf(" ");
         }
     }
-    printf(":");
+    printf(" : ");
     const char *utf8 = xkb_compose_table_entry_utf8(entry);
     if (*utf8 != '\0') {
-        printf(" \"%s\"", utf8);
+        if (!escape_utf8_string(utf8, buf, sizeof(buf))) {
+            fprintf(stderr,
+                    "ERROR: Cannot escape the string: "
+                    "unsufficient buffer size.\n");
+            printf("\"\"");
+        } else {
+            printf(" \"%s\"", buf);
+        }
```

Comment:
Re-using `buf` here feels naughty. Any reason we can't just return an allocated buffer from `escape_utf8_string()` and worry about greater things in life than bufsizes? :)


If you alloc a buffer `4 * strlen(from) + 1` then our string will definitely fit and we can realloc to the smaller size before return.